torch/fx/experimental/symbolic_shapes.py

```diff
@@ -191,6 +191,8 @@ def sym_float(self) -> "SymNode":  # noqa: F811
 
     # Today we error on calling int on a symbolic shape, as this is a very accessible footgun.
     def int_(self):
+        if len(self.expr.free_symbols) == 0:
+            return int(self.expr)
```

Comment:
I don't like this. It's a good hack that I can believe improves coverage, but in general we shouldn't let people call int() on SymInt, because even if it happens to work on the current test, if later someone feeds an actual symbolic error, it will raise an error. This makes it easy to create compile errors that don't show up until later.